pkg/cloud/vsphere/services/loadbalancer/aws/loadbalancers.go

```diff
@@ -0,0 +1,315 @@
+	}
+	describeListnerOutput, err := svc.ELB.DescribeListeners(describeListnerInput)
+
+	if err != nil || len(describeListnerOutput.Listeners) == 0 {
+		if IsNotFound(err) || len(describeListnerOutput.Listeners) == 0 {
+			listenerInput := &elbv2.CreateListenerInput{
+				LoadBalancerArn: loadBalancerArn,
+				Port:            aws.Int64(int64(6443)),
+				Protocol:        aws.String(elbv2.ProtocolEnumTcp),
+			}
+
+			listenerInput.DefaultActions = append(listenerInput.DefaultActions, &elbv2.Action{
+				TargetGroupArn: targetGroupArn,
+				Type:           aws.String(elbv2.ActionTypeEnumForward),
+			})
+
+			listenerOutput, err := svc.ELB.CreateListener(listenerInput)
+			if err != nil {
+				return nil, err
+			}
+			return listenerOutput.Listeners[0].Port, nil
+		}
+		return nil, err
+	}
+
+	modifyListenerInput := &elbv2.ModifyListenerInput{
+		Port:        aws.Int64(int64(6443)),
+		Protocol:    aws.String(elbv2.ProtocolEnumTcp),
+		ListenerArn: describeListnerOutput.Listeners[0].ListenerArn,
+	}
+	modifyListenerInput.DefaultActions = append(modifyListenerInput.DefaultActions, &elbv2.Action{
+		TargetGroupArn: targetGroupArn,
+		Type:           aws.String(elbv2.ActionTypeEnumForward),
+	})
+	modifyListenerOutput, err := svc.ELB.ModifyListener(modifyListenerInput)
+	if err != nil {
+		return nil, err
+	}
+	listenerPort = modifyListenerOutput.Listeners[0].Port
+	return listenerPort, nil
+}
+
+func (svc *Service) deleteLoadBalancer(clusterName string) error {
+
+	describeLoadBalancersInput := &elbv2.DescribeLoadBalancersInput{
+		Names: []*string{aws.String(generateELBName(clusterName))},
+	}
+	describeLoadBalancersOutput, err := svc.ELB.DescribeLoadBalancers(describeLoadBalancersInput)
+	if err != nil {
+		if IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	loadBalancerArn := describeLoadBalancersOutput.LoadBalancers[0].LoadBalancerArn
+	deleteLoadBalancerInput := &elbv2.DeleteLoadBalancerInput{
+		LoadBalancerArn: loadBalancerArn,
+	}
+	_, err = svc.ELB.DeleteLoadBalancer(deleteLoadBalancerInput)
+	return err
+}
+
+func (svc *Service) deleteTargetGroup(clusterName string) error {
+	describeTargetGroupInput := &elbv2.DescribeTargetGroupsInput{}
+	describeTargetGroupInput.Names = append(describeTargetGroupInput.Names, aws.String(clusterName+"-controlPlane"))
+
+	describeTargetGroupOutput, err := svc.ELB.DescribeTargetGroups(describeTargetGroupInput)
+	if err != nil {
+		if IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	deleteTargetGroupInput := &elbv2.DeleteTargetGroupInput{
+		TargetGroupArn: describeTargetGroupOutput.TargetGroups[0].TargetGroupArn,
+	}
+	_, err = svc.ELB.DeleteTargetGroup(deleteTargetGroupInput)
+	return err
+}
+
+// Delete performs an ELB resource deletion
+func (svc *Service) Delete(vsphereCluster *infrav1.VSphereCluster) error {
+	clusterName := vsphereCluster.Name
+	if err := svc.deleteLoadBalancer(clusterName); err != nil {
+		return err
+	}
+
+	if err := svc.deleteTargetGroup(clusterName); err != nil {
```

Comment:
Hi @yastij,

This could be simplified as:

```golang
return svc.deleteTargetGroup(clusterName)
```

Or please feel free to leave as-is if you think we may include additional logic beneath this call one day.